app/src/main/java/org/metabrainz/android/presentation/features/search/SearchPagingSource.kt

```diff
@@ -28,11 +30,16 @@ class SearchPagingSource(val mbEntityType: MBEntityType, val query: String) : Pa
                 else -> (count - offset - pageSize).coerceAtLeast(0)
             }
             // itemsAfter is required to be at least otherwise the current page will be not loaded
+            val nextKey = if (entity.isEmpty)  null
+            else pageSize+offset
 
+            if (offset==0 && entity.isEmpty && count==0){
```

Comment:
Also I put a check if we are getting response or not .
Before if an empty response comes in the first attempt (offset=0) , we were not showing No result animation.
Now we are throwing an exception , so its working fine.